### PR TITLE
fix updateUser in securityController: persist method need a hydrated …

### PR DESCRIPTION
**Pull request by stafyniaksacha**
…user to work

### CODE REVIEW

**Comment by ballinette** [comment]
missing unit test for this, but +1

**Comment by scottinet** [comment]
Is this pull request still relevant?

File: lib/api/controllers/securityController.js [hunk 1]
```diff
@@ -388,7 +390,19 @@ module.exports = function SecurityController (kuzzle) {
 
         return kuzzle.repositories.user.load(newRequestObject.data._id)
           .then(user => {
-            return kuzzle.repositories.user.persist(_.extend(user, newRequestObject.data.body), { database: { method: 'update' } });
```
Comment (benoitvidis) [inline lib/api/controllers/securityController.js]: Isn't `user`already hydrated at this point?

Comment (stafyniaksacha) [inline lib/api/controllers/securityController.js]: Nope, the `Repository.load` method didn't return a real `User` object: it contains data in a `_source` subfield (seems to be the raw output of elastic search) so the `_.extend` didn't work as we want here.

On my way, we need to plan a refactor on the Repositories methods to ensure the validity of theirs return values.

To reproduce the bug, just create a new user with admin profile, update it with anonymous profile. Then get it and see that the profile is switched to default.

Comment (benoitvidis) [inline lib/api/controllers/securityController.js]: Then, if so, the problem lies within the repository code.
The UserRepository::load method [_should_ resolve to a hydrated object](https://github.com/kuzzleio/kuzzle/blob/develop/lib%2Fapi%2Fcore%2Fmodels%2Frepositories%2Frepository.js#L55).

I would rather fix the repository than putting the logic in the controller.

Comment (stafyniaksacha) [inline lib/api/controllers/securityController.js]: Yes, it's the way we are going to.

@j33f is implementing this.

So this PR will not be relevant after.

**Pull request closed without merging**